### PR TITLE
new: cloudflare_argo_tunnels datasource

### PR DESCRIPTION
**Pull request by eidam**
Hey! 👋

I was thinking about writing another argo tunnel example but I would need a `cloudflare_argo_tunnels` data source for that. 
I tried to put that together real quick, I am not a pro when it comes to writing terraform providers, but I am happy to hear your feedback!

### CODE REVIEW

File: cloudflare/data_source_argo_tunnels.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+	if err != nil {
+		return fmt.Errorf("Error setting tunnels: %s", err)
+	}
+
+	d.SetId(stringListChecksum(tunnelIds))
+	return nil
+}
+
+func expandArgoTunnelFilter(d interface{}) (*argoTunnelsearchFilter, error) {
+	cfg := d.([]interface{})
```
Comment (jacobbednarz) [inline cloudflare/data_source_argo_tunnels.go:116]: we shouldn't need to be doing this in here as the API endpoint already [has the option to filter the results](https://api.cloudflare.com/#argo-tunnel-list-argo-tunnels) which is performed on the server side. we will however need to [update the cloudflare-go library](https://github.com/cloudflare/cloudflare-go/blob/e625e78fb682d9f46bfe5ffde938ed19a1eff3e2/argo_tunnel.go#L47) to support this.

**Comment by jacobbednarz** [comment]
other than doing the filtering via the server side calls, i think we're in a pretty good spot with this one.

**Comment by jacobbednarz** [comment]
this is still pending the changes to `cloudflare-go`. @eidam are you interested in fixing that?

**Comment by eidam** [comment]
> this is still pending the changes to `cloudflare-go`. @eidam are you interested in fixing that?

I will do my best to take a look this week. 👍

**Comment by LCedergren** [comment]
Is this still being worked on?

**Comment by jay-bware** [comment]
Any updates when will this be merged??

**Comment by jacobbednarz** [comment]
the filtering capabilities are now available in cloudflare-go - https://github.com/cloudflare/cloudflare-go/blob/dc6c002de52f15566cbcedb5be9a5e758858dabb/tunnel.go#L185-L191.

we'll need to rebase this against the latest `master` branch in order to get it up to date and use all of the latest goodness.

**Comment by jacobbednarz** [comment]
closing this out due to necromancy. if someone would like to grab this, please open a new PR with filtering capabilities from cloudflare-go.

**Pull request closed without merging**